Update readme to refer to v3.1 and link to breaking changes (#2529)
dotnet/winforms: README.md

### Code Coverage

|               | Production Code                                   | 
|-------------  |-------------------------------------------------  |
| master        | [![codecov][master-coverage-prod]][coverage]      |
| release/3.1   | [![codecov][release31-coverage-prod]][coverage]   |
| release/3.0   | [![codecov][release3-coverage-prod]][coverage]    |

## How to Engage, Contribute, and Provide Feedback

Some of the best ways to contribute are to try things out, file bugs, join in design conversations, and fix issues.

* The [contributing guidelines][contributing] and the more general [.NET Core contributing guide](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/contributing.md) define contributing rules.
* The [Developer Guide](Documentation/developer-guide.md) defines the setup and workflow for working on this repository.
* If you have a question or have found a bug, [file an issue](https://github.com/dotnet/winforms/issues/new?template=bug_report.md).
* Use [daily builds][getting-started] if you want to contribute and stay up to date with the team.

### .NET Framework issues

> Issues with .NET Framework, including Windows Forms, should be filed on [VS developer community](https://developercommunity.visualstudio.com/spaces/61/index.html) or [Product Support](https://support.microsoft.com/en-us/contactus?ws=support). They should not be filed on this repository.

Reviewer: ```suggestion
Issues with .NET Framework, including Windows Forms, should be filed on the [Developer Community](https://developercommunity.visualstudio.com/spaces/61/index.html) or [Product Support](https://support.microsoft.com/en-us/contactus?ws=support) websites. They should not be filed on this repository.
```

### Reporting security issues

Security issues and bugs should be reported privately via email to the Microsoft Security Response Center (MSRC) <secure@microsoft.com>. You should receive a response within 24 hours. If for some reason you do not, please follow up via email to ensure we received your original message. Further information, including the MSRC PGP key, can be found in the [Security TechCenter](https://www.microsoft.com/msrc/faqs-report-an-issue). Also see info about related [Microsoft .NET Core and ASP.NET Core Bug Bounty Program](https://www.microsoft.com/msrc/bounty-dot-net-core).

## Relationship to .NET Framework

> This codebase is a fork of the Windows Forms code in the .NET Framework 4.8. In Windows Forms .NET Core 3.0 we have strived to bring the two runtimes to a parity, however since then we have done a number of [changes, including breaking](https://docs.microsoft.com/en-us/dotnet/core/compatibility/winforms), which diverged the two.

Reviewer: ```suggestion
This codebase is a fork of the Windows Forms code in the .NET Framework 4.8. In Windows Forms .NET Core 3.0, we've strived to bring the two runtimes to a parity. However, since then, we've done a number of changes, including [breaking changes](https://docs.microsoft.com/dotnet/core/compatibility/winforms), which diverged the two.
```

> Please refer to [Porting guide][porting-guidelines] to learn more about breaking changes.

Reviewer: ```suggestion
For more information about breaking changes, see the [Porting guide][porting-guidelines].
```
